docs/blueprints/storage-node-satellite-selection.md

```diff
@@ -0,0 +1,199 @@
+```
+
+### Allow List
+
+The Allow List contains Satellite URLS of explicitly trusted Satellites.
+
+### Block List
+
+The Block List contains Satellite URLs of explicitly blocked Satellites.
+
+### Building the List of Trusted Satellite URLs
+
+To build the list of trusted Satellite URLs, the following steps are performed,
+in order:
+
+1. Trusted Satellite Lists are downloaded from all Trusted Satellite List URLs and aggregated _in the order_ they are specified in the configuration.
+1. URLs from the Allow List are then aggregated into the trusted list.
+1. URLs in the Block List are removed from the trust list, if present. URLs
+   in the Block list that are not present in the trust list are ignored.
+
+If a Trusted Satellite List cannot be fetched a warning should be logged. If
+available, the last known copy from the Trusted Satellite List URL should be
+used. Storage Nodes should persist the downloaded lists.
+
+When aggregating Satellite URLs, the following rules **MUST** be followed:
+
+* Satellite URL IDs are equivalent if they are equal, or one is a prefix of the
+  other (including an empty ID).
+* Satellite URLs are equivalent if the address portions are equal.
+* When aggregating equivalent Satellite URLs with equivalent IDs, the Satellite
+  URL with the longer ID is preferred over one with a shorter (or no) ID.
+* When aggregating equivalent Satellite URLs with non-equivalent IDs, the new
+  URL wins.
+* When blocking URLs, ID equivalence is ignored. In other words, URLs in the
+  trust list are equivalent with one in the Block List, they are removed.
+
+#### Examples
+
+##### Longest ID Preferred
+
+List A:
+
+```
+abcd@foo.com:7777
+foo.com:7777
+```
+
+List B:
+
+```
+ab@foo.com:7777
+```
+
+Results in:
+
+```
+abcd@foo.com:7777
+```
+
+##### Last URL Wins When ID is not Equivalent
+
+List A:
+
+```
+abcd@foo.com:7777
+```
+
+List B:
+
+```
+ef@foo.com:7777
+```
+
+Results in:
+
+```
+ef@foo.com:7777
+```
+
+##### URL in the Block List
+
+List A:
+
+```
+abcd@foo.com:7777
+```
+
+Block List:
+
+```
+foo.com:7777
+```
+
+Results in:
+
+```
+
+```
+
+### Refreshing the List
+
+The list of trusted Satellite URLs should be refreshed daily (with some jitter).
+
+### Backwards Compatability
+
+The old piecestore configuration (i.e. `piecestore.OldConfig`) currently contains a
+comma separated list of trusted Satellite URLs (`WhitelistedSatellites`). It
+defaults to the current list of known good satellites. On startup, if the new
+configuration is unset, then the old configuration should be used to form
+a fixed set of trusted Satellite URLs.
+
+## Open Issues
+
+* How long should storage nodes use cached/persisted lists for? Should lists be persisted at all?
+* If aggregation yields no URLs (list URL unreachable) should we default to anything? How should this be reported?
+* If block listing removes all URLs, how should this be reported?
+* Can we safely auto-migrate storage nodes into this new method of management?
+
```

Comment:
What happens if a node operator blocks a satellite that it is storing data for? 
- Does the Node block the satellite and automatically delete the data it was storing for it
- Does the node block the satellite and keep the data it was storing for it and then has to manually delete it later

When a node blocks a satellite it will simply stop accepting any and all requests from it? 

Does a node tell a satellite it is blocking it?

Reply:
The piecestore rejects operations from satellites that the Storage Node does not trust. From what I can tell, garbage collection is driven by the Satellite, so there might need to be a way to eventually reclaim data previously stored for a now trusted Satellite. I'll put that under Open Issues since that falls slightly out of scope for this particular design and is already the case today.

Is there utility in telling a satellite it is being blocked? Is there a feature you have in mind that would be lit up by that?

Reply:
I suppose the only reason I made the comment about the node telling the satellites its blocking it was so that the satellite can stop choosing the node for operations such as uploads, downloads, and audits on the network. 
the data the node was holding would also not be accessible by the satellite so it would be good for the satellite to know that so that it can make sure it has an accutre count of segments for repair purposes.

Reply:
I see. There is a specific error message that is returned by satellites in this case (returns PermissionDenied). Maybe this is enough of a signal to Satellites?